velox/type/tests/TypeTest.cpp

```diff
@@ -117,6 +117,24 @@ TEST(Type, Date) {
   EXPECT_EQ(date->begin(), date->end());
 }
 
+TEST(Type, Decimal) {
+  auto shortDecimal = SHORT_DECIMAL(10, 5);
+  EXPECT_EQ(shortDecimal->toString(), "SHORT_DECIMAL(10,5)");
+  EXPECT_EQ(shortDecimal->size(), 0);
+  EXPECT_THROW(shortDecimal->childAt(0), std::invalid_argument);
+  EXPECT_EQ(shortDecimal->kind(), TypeKind::SHORT_DECIMAL);
+  EXPECT_STREQ(shortDecimal->kindName(), "SHORT_DECIMAL");
+  EXPECT_EQ(shortDecimal->begin(), shortDecimal->end());
+
+  auto longDecimal = LONG_DECIMAL(10, 5);
+  EXPECT_EQ(longDecimal->toString(), "LONG_DECIMAL(10,5)");
+  EXPECT_EQ(longDecimal->size(), 0);
+  EXPECT_THROW(longDecimal->childAt(0), std::invalid_argument);
+  EXPECT_EQ(longDecimal->kind(), TypeKind::LONG_DECIMAL);
+  EXPECT_STREQ(longDecimal->kindName(), "LONG_DECIMAL");
+  EXPECT_EQ(longDecimal->begin(), longDecimal->end());
```

Comment:
> Is it possible to create a variant of decimal type? If so, would you add a test to demonstrate that?

Are there any existing tests for reference?



> It would be nice to also add some basic tests for flat, dictionary and constant vectors of the decimal types. These may go to velox/vector/tests/VectorTest.cpp

Can this be a separate PR.

Reply:
It is preferable to add tests in the same PR as new functionality.

Reply:
As discussed in the meeting. This PR will only focus on adding a new Decimal Type. The variant changes will be made on top of Prestissimo grammar changes.